mne/gui/_coreg.py

```diff
@@ -226,10 +228,36 @@ def _get_default(var, val):
         self._renderer = _get_renderer(
             size=self._defaults["size"], bgcolor=self._defaults["bgcolor"])
         self._renderer.enable_depth_peeling()
-        self._renderer._window_close_connect(self._clean)
         self._renderer.set_interaction(interaction)
         self._renderer._status_bar_initialize()
 
+        # connect callback to close event
+        def closeEventCallback():
+            from ..viz.backends.renderer import MNE_3D_BACKEND_TESTING
+            if not self._trans_saved:
+                from PyQt5.QtWidgets import QMessageBox
+                if MNE_3D_BACKEND_TESTING:
+                    ret = QMessageBox.Cancel
+                else:
+                    ret = QMessageBox.warning(
+                        self._renderer._window,
+                        "CoregistrationUI",
+                        "The Head<>MRI transform has not been saved. "
+                        "Do you want to save it?",
+                        QMessageBox.Save | QMessageBox.Cancel,
+                        QMessageBox.Save
+                    )
+                if ret == QMessageBox.Save:
+                    self._forward_widget_command(
+                        "save_trans", "set_value", None)
+                else:
+                    assert ret == QMessageBox.Cancel
+            if not self._fids_saved:
+                self._forward_widget_command(
+                    "save_mri_fids", "set_value", None)
```

Comment:
I did not find a nice way to integrate that in the GUI API. That is very specific after all